src/imitation/rewards/reward_nets.py

```diff
@@ -318,6 +302,210 @@ def forward(self, state, action, next_state, done):
         return outputs
 
 
+class NormalizedRewardNet(RewardNetWrapper):
+    """A reward net that normalizes the output of its base net.
+
+    Only requires the implementation of a forward pass (calculating rewards given
+    a batch of states, actions, next states and dones) as it inherits RewardNet.
+    """
+
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        base: RewardNet,
+        rew_normalize_class: Type[nn.Module],
+        normalize_images: bool = True,
+    ):
+        """Initialize the NormalizedRewardNet.
+
+        Args:
+            observation_space: the observation space of the environment
+            action_space: the action space of the environment
+            base: a base RewardNet
+            rew_normalize_class: The class to use to normalize rewards. This
+                can be any nn.Module that preserves the shape; e.g. `nn.Identity`,
+                `nn.BatchNorm*`, `nn.LayerNorm`, or `networks.RunningNorm`.
+            normalize_images: passed through to `RewardNet.__init__`,
+                see its documentation
+        """
+        super().__init__(
+            observation_space=observation_space,
+            action_space=action_space,
+            base=base,
+            normalize_images=normalize_images,
+        )
+        # assuming reward is always a scalar
+        self.rew_normalize_layer = (
+            rew_normalize_class(1) if rew_normalize_class else None
+        )
+
+    def predict_processed(
```

Comment:
This implementation looks like it should work but I think we can maybe come up with something a bit cleaner.

It's a shame you have to convert `rew` to Tensor `rew_th`, then go back again to NumPy. `predict` already has the tensor input, but it converts it to `NumPy` before returning.

Since we can make changes to base `RewardNet` class, one option is to just move most of `predict` into something like `predict_tensor` and then `predict` just calls `predict_tensor` and does the final conversion to NumPy: https://github.com/HumanCompatibleAI/imitation/blob/53d8ab17e4ef116695f73fcae673d1ddb79242ac/src/imitation/rewards/reward_nets.py#L141 This might actually be useful for other users anyway. WDYT?

Reply:
make sense